controllers/AuthenticationController/createUser.js

```diff
@@ -1,11 +1,27 @@
 const User = require("../../models/Users");
+const Role = require("../../models/Role"); 
 const bcrypt = require("bcrypt");
 
 const createUser = async (req, res, next) => {
-  const { firstName, lastName, email, password } = req.body;
+  const { firstName, lastName, email, password, roleName } = req.body;
   const hashedPassword = bcrypt.hashSync(password, 10);
 
   try {
+    let roleId; // Initialize roleId variable
+
+    if (roleName) {
+      // If a role name is provided, find the role by name
+      const role = await Role.findOne({ where: { name: roleName } });
+
+      if (role) {
+        roleId = role.id;
+      } else {
+        // If the role doesn't exist, create a new role with the provided name
+        const newRole = await Role.create({ name: roleName });
+        roleId = newRole.id;
```

Comment:
No need to create a new role, return an error saying role doesn't exist instead.